plugins/module_utils/network/sonic/config/aaa/aaa.py

```diff
@@ -128,12 +129,14 @@ def set_state(self, want, have):
 
         if state == 'deleted':
             commands = self._state_deleted(want, have)
-        elif state == 'merged':
+        elif state == 'merged' or state == 'replaced':
             diff = self.get_diff_aaa(want, have)
-            commands = self._state_merged(want, have, diff)
+            commands = self._state_merged(diff)
+        elif state == 'overridden':
+            commands = self._state_overridden(want, have)
         return commands
 
-    def _state_merged(self, want, have, diff):
+    def _state_merged(self, diff):
```

Comment:
This function used by "merged" and "replaced". for example, have.default_auth = ['local', 'ldap'] and want.default_auth = ['local',  'radius']. If the required result is as folloing:  for "merged", the result have..default_auth = ['local', 'ldap',  'radius'], and for "replaced", the result have..default_auth = ['local',  'radius'], this function can not be used for both states.

Line#151: for "replaced", the state should be updated to "replaced".

Reply:
The behavior is not such in SONiC code. A patch on the authentication-method list does not append to the list like expected. Instead the entire list is replaced.  I had filed a bug for this but Eric told me that is the desired behavior.

I went ahead and created a separate method for replaced to update the state.

Reply:
Good to know. Thanks!